src/vendor/core/warning.js

```diff
@@ -31,6 +31,10 @@ if (__DEV__) {
       );
     }
 
+    if (format.indexOf('Failed CompositeComponent proptype check. ') === 0) {
+      return; // Ignore CompositeComponent proptype check.
+    }
+
```

Comment:
Note: you can use this fiddle to start from for latest React: http://jsfiddle.net/reactjs/ghmpo42k/ - that will use the latest builds from master instead of stable releases.